packages/babel-plugin-emotion/src/index.js

```diff
@@ -495,19 +496,43 @@ export default function(babel: Babel) {
         exit(path: BabelPath, state: EmotionBabelPluginPass) {
           if (state.staticRules.length !== 0) {
             const toWrite = state.staticRules.join('\n').trim()
-            const filenameArr = path.hub.file.opts.filename.split('.')
-            filenameArr.pop()
-            filenameArr.push('emotion', 'css')
-            const cssFilename = filenameArr.join('.')
-            const exists = fs.existsSync(cssFilename)
-            addSideEffect(path, './' + nodePath.basename(cssFilename))
+            let cssFilename = path.hub.file.opts.sourceFileName
```

Comment:
I think if you replace `path.hub.file.opts.sourceFileName` with `path.hub.file.opts.generatorOpts ? path.hub.file.opts.generatorOpts.sourceFileName : path.hub.file.opts.sourceFileName` the babel 7 error will be fixed.